inspect-image/main.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+func mainE() error {
+	imageName := "harbor-repo.vmware.com/dockerhub-proxy-cache/library/busybox:1.33.1"
```

Comment:
This is a test you can run from this branch to see the way I found out how all this works.

At the end I figure out:
- The format the original manifest was using for its JSON.
- The original digest matched one computed here by sha256-summing that preformatted JSON.
- I could turn the manifest object back to JSON and format it again to obtain the original JSON and digest.
- I managed to also rebuild a JSON manifest from a loaded image so that it would render the expected digest.

So the final output was:
~~~
Digest:               febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b
Recomputed digest:    febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b
Reconstructed digest: febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b
Reloaded Digest:      fec32035de6529ae53db1cea4a4885f9f980014a6a719535a49c23961d036035
Rebuilt digest:       febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b
~~~
The reloaded digest is the only one different, as it is before fixing the JSON format.

I could turn this into a unit test if this kind of fix is the path we finally decide upon.